lib/fauna/query.rb

```diff
@@ -0,0 +1,249 @@
+    # Reference: {FaunaDB Sets}[https://faunadb.com/documentation#queries-sets]
+    def self.match(terms, index_ref)
+      Set.new(terms, index_ref).to_hash
+    end
+
+    # A union expression
+    #
+    # Reference: {FaunaDB Sets}[https://faunadb.com/documentation#queries-sets]
+    def self.union(sets)
+      { 'union' => sets }
+    end
+
+    # An intersection expression
+    #
+    # Reference: {FaunaDB Sets}[https://faunadb.com/documentation#queries-sets]
+    def self.intersection(sets)
+      { 'intersection' => sets }
+    end
+
+    # A difference expression
+    #
+    # Reference: {FaunaDB Sets}[https://faunadb.com/documentation#queries-sets]
+    def self.difference(source, sets)
+      { 'difference' => sets.unshift(source) }
+    end
+
+    # A join expression
+    #
+    # Reference: {FaunaDB Sets}[https://faunadb.com/documentation#queries-sets]
+    def self.join(source, target)
+      { 'join' => source, 'with' => target }
+    end
+
+    # :section: Miscellaneous Functions
+
+    # An equals function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.equals(values)
+      { 'equals' => values }
+    end
+
+    # A concat function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.concat(strings)
+      { 'concat' => strings }
+    end
+
+    # A contains function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.contains(path, value)
+      { 'contains' => path, 'in' => value }
+    end
+
+    # A select function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.select(path, data, params = {})
+      append_params({ 'select' => path, 'from' => data }, params, %w(ts after before size events sources))
+    end
+
+    # An add function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.add(numbers)
+      { 'add' => numbers }
+    end
+
+    # A multiply function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.multiply(numbers)
+      { 'multiply' => numbers }
+    end
+
+    # A subtract function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.subtract(numbers)
+      { 'subtract' => numbers }
+    end
+
+    # A divide function
+    #
+    # Reference: {FaunaDB Miscellaneous Functions}[https://faunadb.com/documentation#queries-misc_functions]
+    def self.divide(numbers)
+      { 'divide' => numbers }
+    end
+
+  private
+
+    def self.append_params(source, params, allowed)
+      source.merge(params.select { |key, value | allowed.include?(key) && !value.nil? })
```

Comment:
`allowed` doesn't seem to serve much purpose but to potentially confuse people when parts of their query go missing.
Also, don't need that space after "value"